application/server/server.go

```diff
@@ -466,22 +467,36 @@ func textDocumentDidSaveHandler() jrpc2.Handler {
 		bgCtx := context.Background()
 		logger := log.With().Str("method", "TextDocumentDidSaveHandler").Logger()
 
-		logger.Info().Interface("params", params).Msg("Receiving")
-		di.FileWatcher().SetFileAsSaved(params.TextDocument.URI)
 		filePath := uri.PathFromUri(params.TextDocument.URI)
 
+		logger.Info().
+			Str("documentURI", filePath).
+			Msg("Received save event for document")
+
+		di.FileWatcher().SetFileAsSaved(params.TextDocument.URI)
+
 		// todo can we push cache management down?
 		f := workspace.Get().GetFolderContaining(filePath)
 		autoScanEnabled := config.CurrentConfig().IsAutoScanEnabled()
+
 		if f != nil && autoScanEnabled {
-			f.ClearDiagnosticsFromFile(filePath)
-			di.HoverService().DeleteHover(filePath)
+			logger.Info().
+				Str("documentURI", filePath).
+				Msg("Starting scan for saved document")
+
+			// f.ClearDiagnosticsFromFile(filePath)
+			// di.HoverService().DeleteHover(filePath)
+
 			go f.ScanFile(bgCtx, filePath)
 		} else {
-			if autoScanEnabled {
-				logger.Warn().Str("documentURI", filePath).Msg("Not scanning, file not part of workspace")
-			} else {
-				logger.Warn().Msg("Not scanning, auto-scan is disabled")
+			if f == nil {
```

Comment:
suggestion: pull this up as a guard expression, so we don't have these deep hierarchies and simplify the ifs.